### ISSUE TITLE
Improve visibility of inactive customer profile tabs

### ISSUE THREAD

**Issue by mindctrl** [easydigitaldownloads/easy-digital-downloads#3791]
The inactive tabs are too light and don't have enough contrast.

![image](https://cloud.githubusercontent.com/assets/1254870/9772913/178e8f0c-570d-11e5-8409-0e4b81112681.png)

Using a color contrast checker that checks for accessibility issues ([link](http://webaim.org/resources/contrastchecker/)), the colors fail for not having enough contrast. I think we just need to darken the icon color.

**Comment by chriscct7** [comment]
Cross ref with FES when this goes in please, so we can make sure to keep the CSS consistent :)

**Comment by zackkatz** [comment]
Not to derail this issue, but I'd also suggest adding labels with the logos for easier identification of what they do. I never know exactly what the tabs mean at a glance, even though hovering shows it.

It looks way worse, but I'd argue it's easier to use.
 
<img width="312" alt="screen shot 2015-09-13 at 5 17 28 pm" src="https://cloud.githubusercontent.com/assets/870979/9839566/692da07e-5a3b-11e5-9746-4b6afa319c94.png">
…

**Comment by chriscct7** [comment]
I completely agree. Particularly as extensions start to register custom settings.

**Comment by chriscct7** [comment]
This should probably be a seperate issue but there's *alot* of effort involved in re-using these profiles across extensions.

**Comment by pippinsplugins** [comment]
We had extensive conversations on this when first introducing custom details page. See #2548

**Comment by chriscct7** [comment]
That discussion starts here: https://github.com/easydigitaldownloads/Easy-Digital-Downloads/issues/2548#issuecomment-77177679 (so you don't have to wade through that ticket).

I think we should really reconsider this decision. I really like the proposal above or the one John did on the other ticket

**Comment by pippinsplugins** [comment]
If we decide to go with horizontal tabs, I think we should go for something closer to the WooCommerce product configuration tabs UI.

**Comment by spencerfinnell** [comment]
I definitely prefer the WooCommerce style. For having it contained in a meta box, and the tabs are a lot clearer (and easier to handle on small devices).

**Comment by pippinsplugins** [comment]
Note, color of inactive tabs has been fixed in #4443

**Comment by cklosowski** [comment]
I vote that we move the Contains Metabox with vertical tabs containing text. I think this is a route that people are comfortable with

**Comment by cklosowski** [comment]
Moving to 2.9. While it would be nice to have this done, we have some more pressing enhancements right now. If someone jumps on this before I get to it, I think we should look at the WooCommerce tabbed box model as a guide, I think that would suit here well.

**Comment by SDavisMedia** [comment]
I took a stab at this. Few things to note:

* It's backwards compatible, so it won't disrupt anything already in the wild. Likewise, it does not affect card views that do not use tabs. That means Recurring Payments card view is still good and FES rebuilt the entire vendor view on its own, so it's unaffected.
* Software Licensing will need very minor changes before SL 3.5 goes out. I'll take care of that. (Cc: @ghost1227) 
* Though it now displays text on the tab, it only _shows_ the text that people need to _see_. The word `Customer ` is prepended to all tab titles but wrapped in the screen reader class... that way everything is optimized for _all_ users. Please review the code and let me know if you have beef with how it's done... particularly the part where I make sure no extension tries to add a tab title with "Customer" already at the beginning of the string, like EDD Wallet.
* It's fully responsive.
* Das it mane.

<img width="872" alt="screenshot 2016-11-18 12 38 59" src="https://cloud.githubusercontent.com/assets/2359131/20447087/d556fdb8-ad91-11e6-876f-09b64ee303bc.png">

<img width="661" alt="screenshot 2016-11-18 12 39 56" src="https://cloud.githubusercontent.com/assets/2359131/20447093/dbb9172c-ad91-11e6-987f-5ff2463595d1.png">

<img width="411" alt="screenshot 2016-11-18 12 40 18" src="https://cloud.githubusercontent.com/assets/2359131/20447097/e2220f92-ad91-11e6-9956-01b9e0794c91.png">

**Comment by pippinsplugins** [comment]
I _love_ this.

Do these changes carry over properly to Recurring Payments and FES or will those need updates too?

**Comment by ghost1227** [comment]
@SDavisMedia Gorgeous! I'll make sure to update Wallet whenever this nightmare of an update is ready.

**Comment by pippinsplugins** [comment]
Missed comments about RP and FES above. Good stuff :+1:

**Comment by pippinsplugins** [comment]
Moved back to 2.7 since @SDavisMedia has done all the heavy lifting already.

**Comment by chriscct7** [comment]
What happens when the text overflows, such as when an extension makes a tab called "my awesome but super long tab name"

**Comment by mindctrl** [comment]
@chriscct7 we exile said extension.

**Comment by chriscct7** [comment]
I know you're joking but just wanted to point out another way text could get that long inadvertently is via translations where 1 word could become a 30 character word in another language

**Comment by ghost1227** [comment]
Joking aside, it's a valid question.

<img width="163" alt="screen shot 2016-11-21 at 10 30 51 am" src="https://cloud.githubusercontent.com/assets/36433/20491022/a30fe9e6-afd5-11e6-8714-f0dbc22b6863.png">

**Comment by mindctrl** [comment]
Here's what Woo does on their product edit page:

![image](https://cloud.githubusercontent.com/assets/1254870/20491792/80b30544-afe0-11e6-834d-ddbec289a583.png)

I think that looks fine, and it looks like what @ghost1227's screenshot looks like.

**Comment by spencerfinnell** [comment]
Agreed. Only thing that could be done is keeping the next line left-aligned with the first, but I don't think that is necessary either.

**Comment by SDavisMedia** [comment]
I'll pretty up the text wrap but yes, it just needs to wrap. We won't stop that from happening. I'll figure something out for a really long single word, @chriscct7, because that's a totally valid concern. I'll weigh some pros and cons for the tab width.

@spencerfinnell Thanks for the code comment. I'll address that.

**Comment by chriscct7** [comment]
@SDavisMedia the simple solution might just be to `break-word` it.

**Comment by SDavisMedia** [comment]
@chriscct7 Yup. My contribution to Twenty Sixteen was for the same exact use case. :+1:

**Comment by SDavisMedia** [comment]
Check this out, @spencerfinnell: https://github.com/easydigitaldownloads/easy-digital-downloads/commit/6848de09dfb55e1e658e5e6abc7b3eefd6bea084

**Comment by pippinsplugins** [comment]
This looks wonderful. Noticed one small issue with the Software Licensing update, but I logged that https://github.com/easydigitaldownloads/EDD-Software-Licensing/issues/667

**Comment by cklosowski** [comment]
This looks good, althoguh we have a few updates to make to Software Licensing after https://github.com/easydigitaldownloads/EDD-Software-Licensing/issues/667. The names of the tabs aren't showing, so we'll have to get an update out after 2.7 that adds those in there.

**Comment by SDavisMedia** [comment]
@cklosowski nope everything is actually ready to go. Pippin just missed my note about the SL adjustment depending on an EDD version check. The names don't show because EDD `issue/3791` does not have version 2.7 set. No matter which one is released first, EDD or SL, this will work as expected in both EDD and SL.

See: https://github.com/easydigitaldownloads/EDD-Software-Licensing/issues/667#issuecomment-265633721

**Comment by cklosowski** [comment]
:+1: awesome thanks for the details.

On Jan 6, 2017 5:09 PM, "Sean Davis" <notifications@github.com> wrote:

@cklosowski <https://github.com/cklosowski> nope everything is actually
ready to go. Pippin just missed my note about the SL adjustment depending
on an EDD version check. The names don't show because EDD issue/3791 does
not have version 2.7 set. No matter which one is released first, EDD or SL,
this will work as expected in both EDD and SL.

See: easydigitaldownloads/EDD-Software-Licensing#667 (comment)
<https://github.com/easydigitaldownloads/EDD-Software-Licensing/issues/667#issuecomment-265633721>

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/easydigitaldownloads/easy-digital-downloads/issues/3791#issuecomment-271043816>,
or mute the thread

**Comment by pippinsplugins** [comment]
Merged!
